google-cloud-spanner/src/test/java/com/google/cloud/spanner/spi/v1/GfeLatencyTest.java

```diff
@@ -0,0 +1,346 @@
+        assumeTrue(
+                "Skip tests when emulator is enabled as this test interferes with the check whether the emulator is running",
+                System.getenv("SPANNER_EMULATOR_HOST") == null);
+
+        mockSpanner = new MockSpannerServiceImpl();
+        mockSpanner.setAbortProbability(0.0D); // We don't want any unpredictable aborted transactions.
+        mockSpanner.putStatementResult(MockSpannerServiceImpl.StatementResult.query(SELECT1AND2, SELECT1_RESULTSET));
+        mockSpanner.putStatementResult(MockSpannerServiceImpl.StatementResult.update(UPDATE_FOO_STATEMENT, 1L));
+        address = new InetSocketAddress("localhost", 0);
+        server =
+                NettyServerBuilder.forAddress(address)
+                        .addService(mockSpanner)
+                        .intercept(
+                                new ServerInterceptor() {
+                                    @Override
+                                    public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
+                                            ServerCall<ReqT, RespT> serverCall,
+                                            Metadata headers,
+                                            ServerCallHandler<ReqT, RespT> serverCallHandler) {
+                                        return serverCallHandler.startCall(
+                                                new ForwardingServerCall.SimpleForwardingServerCall<ReqT, RespT>(serverCall) {
+                                                    @Override
+                                                    public void sendHeaders(Metadata headers){
+                                                        headers.put(
+                                                                Metadata.Key.of("server-timing", Metadata.ASCII_STRING_MARSHALLER),
+                                                                String.format("gfet4t7; dur=%d", fakeServerTiming.get()));
+                                                        super.sendHeaders(headers);
+                                                    }
+                                                }, headers);
+                                    }
+                                })
+                        .build()
+                        .start();
+        optionsMap.put(SpannerRpc.Option.CHANNEL_HINT, 1L);
+        spanner = createSpannerOptions(address,server).getService();
+        databaseClient =
+                spanner.getDatabaseClient(DatabaseId.of(projectId, instanceId, databaseId));
+
+
+        mockSpannerNoHeader = new MockSpannerServiceImpl();
+        mockSpannerNoHeader.setAbortProbability(0.0D);
+        mockSpannerNoHeader.putStatementResult(MockSpannerServiceImpl.StatementResult.query(SELECT1AND2, SELECT1_RESULTSET));
+        mockSpannerNoHeader.putStatementResult(MockSpannerServiceImpl.StatementResult.update(UPDATE_FOO_STATEMENT, 1L));
+        addressNoHeader = new InetSocketAddress("localhost", 1025);
```

Comment:
Can you try to use port 0 here again? Port 0 means that the system will try to pick up an ephemeral port to bind the server to (https://docs.oracle.com/javase/7/docs/api/java/net/InetSocketAddress.html#InetSocketAddress(java.net.InetAddress,%20int))